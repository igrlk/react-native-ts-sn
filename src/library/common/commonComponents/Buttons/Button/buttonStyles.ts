import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	button: {
		backgroundColor: '$colorGreen',
		width: '100%',
		borderRadius: 5,
		height: 40,
		display: 'flex',
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'center',
	},
	buttonText: {
		color: '$colorWhite',
		fontSize: 14,
		fontFamily: 'Poppins',
	},
	buttonInfo: {
		borderColor: '#E0E0E0',
		borderWidth: 1,
		backgroundColor: '$colorWhite',
	},
	buttonInfoText: {
		color: '#9E9E9E',
	},
	buttonMargin: {
		marginRight: 5,
	},
});
