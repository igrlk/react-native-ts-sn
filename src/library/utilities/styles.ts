import EStyleSheet from 'react-native-extended-stylesheet';

EStyleSheet.build({
	$colorWhite: '#FFFFFF',
	$colorGreen: '#3BB54A',
	$colorBlue: '#0075FF',
	$colorPurple: '#9352DA',
	$colorOrange: '#FF9A40',
	$colorRed: '#EA4640',
	$colorBlack: '#000000',
	$colorGrey: '#333333',
});
