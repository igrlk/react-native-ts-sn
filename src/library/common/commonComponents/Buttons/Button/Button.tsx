import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

import styles from './buttonStyles';

export type ButtonType = 'info' | 'common';

interface ButtonProps {
	children: React.ReactNode;
	onClick: () => void;
	style?: { [key: string]: any };
	type: ButtonType;
	icon?: any;
}

Button.defaultProps = {
	type: 'common',
};

export default function Button({ children, onClick, style: extraStyle, type, icon }: ButtonProps) {
	return (
		<TouchableOpacity
			onPress={onClick}
			style={[styles.button, getStylesForWrapperByType(type), extraStyle]}
		>
			{icon ? <View style={styles.buttonMargin}>{icon}</View> : null} 
			<Text style={[styles.buttonText, getStylesForTextByType(type)]}>{children}</Text>
		</TouchableOpacity>
	);
}

export function getStylesForWrapperByType(type: ButtonType) {
	if (type === 'info') {
		return styles.buttonInfo;
	}
}

export function getStylesForTextByType(type: ButtonType) {
	if (type === 'info') {
		return styles.buttonInfoText;
	}
}
