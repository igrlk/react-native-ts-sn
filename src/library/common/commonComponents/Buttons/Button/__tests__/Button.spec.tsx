import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button, { getStylesForWrapperByType, getStylesForTextByType } from '../Button';
import styles from '../buttonStyles';

describe('Button', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<Button onClick={jest.fn()}>Some text in button</Button>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should call navigator.goBack by clicking on touchable opacity', () => {
		const onClick = jest.fn();
		const component = shallow(<Button onClick={onClick}>Text</Button>);
		component.find('TouchableOpacity').simulate('press');
		expect(onClick).toHaveBeenCalled();
	});
});

describe('getStylesForWrapperByType', () => {
	it('should return styles.buttonInfo if type === info, otherwise undefined', () => {
		expect(getStylesForWrapperByType('info')).toEqual(styles.buttonInfo);
		expect(getStylesForWrapperByType('common')).toBe(undefined);
	});
});

describe('getStylesForTextByType', () => {
	it('should return styles.buttonInfoText if type === info, otherwise undefined', () => {
		expect(getStylesForTextByType('info')).toEqual(styles.buttonInfoText);
		expect(getStylesForTextByType('common')).toBe(undefined);
	});
});
