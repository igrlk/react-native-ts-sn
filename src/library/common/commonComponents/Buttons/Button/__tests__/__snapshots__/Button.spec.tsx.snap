// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    Some text in button
  </Text>
</View>
`;
